programs/protocol/src/actions/forwarder/buy_shipment.rs

```diff
@@ -25,24 +25,37 @@ pub struct BuyShipment<'info> {
     pub forwarder: AccountLoader<'info, Forwarder>,
     #[account(mut)]
     pub signer: Signer<'info>,
+    #[account(mut,
+        constraint = shipment_owner.key() == shipment.load().unwrap().shipper @ Error::InvalidShipperAccount,
+    )]
+    /// CHECK: The account of the shipper
+    pub shipment_owner: AccountInfo<'info>,
     pub system_program: Program<'info, System>,
 }
 
 pub fn handler(ctx: Context<BuyShipment>) -> Result<()> {
     let forwarded = &mut ctx.accounts.bought.load_init()?;
     let shipment = &mut ctx.accounts.shipment.load_mut()?;
-    let shipper = &mut ctx.accounts.shipper.load_mut()?;
     let forwarder = &mut ctx.accounts.forwarder.load_mut()?;
 
-    shipment.owner = *ctx.accounts.signer.key;
+    // Check if the shipment is for sale
+    require_neq!(shipment.forwarder, shipment.shipper , Error::ShipmentNotForSale);
+    require_eq!(shipment.forwarder, Pubkey::default(), Error::ShipmentSold);
+    require_eq!(shipment.carrier, Pubkey::default(), Error::ShipmentSold);
+
+    // Update owner
+    shipment.forwarder = forwarder.creator;
+
+    // Check if the buyer has enough funds
+    require_gte!(ctx.accounts.signer.get_lamports(), shipment.price, Error::NotEnoughFunds);
 
-    **forwarded = BoughtShipment {
-        creator: shipper.creator,
-        buyer: ctx.accounts.signer.key(),
-        owner: ctx.accounts.signer.key(),
+    // Create forwarded shipment
+    **forwarded = ForwardedShipment {
+        forwarder: forwarder.creator,
+        shipment: ctx.accounts.shipment.key(),
+        resell_price: u64::MAX,
```

Comment:
why introduce it if its placeholder

Reply:
seems like might be useful on the frontend

Reply:
how

Reply:
to not change it everywhere when it is implemented